addons/web_editor/static/src/js/editor/odoo-editor/test/spec/copyPaste.test.js

```diff
@@ -768,6 +761,159 @@ describe('Paste', () => {
+            it('should split p with <br> into seperate p elements', async () => {
+                await testEditor(BasicEditor, {
+                    contentBefore: '<p>[]<br></p>',
+                    stepFunction: async editor => {
+                        await pasteHtml(editor, '<p>abc<br>def<br>ghi<br>jkl</p>');
+                    },
+                    contentAfter: '<p>abc</p><p>def</p><p>ghi</p><p>jkl[]</p>',
+                });
+                await testEditor(BasicEditor, {
+                    contentBefore: '<p>[]<br></p>',
+                    stepFunction: async editor => {
+                        await pasteHtml(editor, '<p>abc<br>def<br>ghi<br>jkl</p><p>mno</p>');
+                    },
+                    contentAfter: '<p>abc</p><p>def</p><p>ghi</p><p>jkl</p><p>mno[]</p>',
+                });
+                await testEditor(BasicEditor, {
+                    contentBefore: '<p>[]<br></p>',
+                    stepFunction: async editor => {
+                        await pasteHtml(editor, '<p>abc<br>def<br>ghi<br>jkl</p><p><br></p><p>mno</p>');
+                    },
+                    contentAfter: '<p>abc</p><p>def</p><p>ghi</p><p>jkl</p><p><br></p><p>mno[]</p>',
+                });
+                await testEditor(BasicEditor, {
+                    contentBefore: '<p>[]<br></p>',
+                    stepFunction: async editor => {
+                        await pasteHtml(editor, '<p>abc<br>def<br><br><br>ghi</p>');
+                    },
+                    contentAfter: '<p>abc</p><p>def</p><p><br></p><p><br></p><p>ghi[]</p>',
+                });
+            });
+            it('should split multiple elements with <br>', async () => {
+                await testEditor(BasicEditor, {
+                    contentBefore: '<p>[]<br></p>',
+                    stepFunction: async editor => {
+                        await pasteHtml(editor, '<p>abc<br>def</p><h1>ghi<br>jkl</h1><h2><br></h2><h3>mno<br>pqr</h3>');
+                    },
+                    contentAfter: '<p>abc</p><p>def</p><h1>ghi</h1><h1>jkl</h1><h2><br></h2><h3>mno</h3><p>pqr[]</p>',
+                });
+            });
+            it('should split div with <br>', async () => {
+                await testEditor(BasicEditor, {
+                    contentBefore: '<p>[]<br></p>',
+                    stepFunction: async editor => {
+                        await pasteHtml(editor, '<div>abc<br>def</div>');
+                    },
+                    contentAfter: '<p>abc</p><p>def[]</p>',
+                });
+            });
+            it('should split <br> elements with empty lines', async () => {
+                await testEditor(BasicEditor, {
+                    contentBefore: '<p>[]<br></p>',
+                    stepFunction: async editor => {
+                        await pasteHtml(editor, '<p><br><br><br></p>');
+                    },
+                    contentAfter: '<p><br></p><p><br></p><p><br>[]<br></p>',
```

Comment:
@Zynton You might not like this two br's at end here, but it will be fixed to contentAfter: `(<p><br></p><p><br></p><p>[]<br></p>)` with https://github.com/odoo/odoo/pull/145220

Reply:
Didn't we establish [here](https://github.com/odoo/odoo/pull/136743#discussion_r1492538848) that 3 line breaks in the clipboard means we should insert 3 visible line breaks in the end? We established that for `<p>a<br><br><br></p>` but not specifically for `<p><br><br><br></p>` so I'm not 100% sure. But if that's the case, this is wrong since you're saying the other PR will remove the extra `<br>` and leave us short of one line break.

Reply:
So I tested it again on bare contenteditable [here](https://jsfiddle.net/f3gx5vzp/), and we get `<p><br><br><br></p>` when we copy the content from the start of fourth line meaning
`<p>`
`  [<br>`
`  <br>`
`  <br>`
`  ]<br>`
`</p>`(I copied 4 lines). So right now i was breaking `<p><br><br><br></p>` to `<p><br></p><p><br></p><p><br></p>` rather I should break it to `<p><br></p><p><br></p><p><br></p><p><br></p>`. So the contentAfter would be `<p><br></p><p><br></p><p><br></p><p>[]<br></p>`. Also I'll bring the [commit](https://github.com/odoo/odoo/pull/145220/commits/4df5277c1d17963f1639a500edabb96e0d82be7c) I referenced from another [PR](https://github.com/odoo/odoo/pull/145220) here so its all at one place.

Reply:
Yes. You have 1 paragraph and you're inserting 3 new paragraphs so you should have 4 at the end.

@dmo-odoo @sebgeelen if you want to confirm...

Reply:
In case 2 we want to add 3 line breaks right? But here we only add 2 since `<p>a<br><br><br></p>` only shows 2 visible lines after the letter "a":
![image](https://github.com/odoo/odoo/assets/24205914/1f7ecb00-993c-4103-ad67-38339735a634)
But the version with paragraph breaks is correct.

In case 3 I'm afraid the same is true.

Reply:
I'm not sure I understand what you mean so we'll need to discuss it together on monday.